spec/lib/reporting/authentication_report_spec.rb

```diff
@@ -70,4 +70,75 @@
       end
     end
   end
+
+  describe '#cloudwatch_client' do
+    let(:opts) { {} }
+    let(:subject) { described_class.new(issuer:, time_range:, **opts) }
+    let(:default_args) do
+      {
+        num_threads: 5,
+        ensure_complete_logs: true,
+        slice_interval: 3.hours,
+        progress: false,
+        logger: nil,
+      }
+    end
+
+    describe 'when all args are default' do
+      it 'creates a client with the default options' do
+        expect(Reporting::CloudwatchClient).to receive(:new).with(default_args)
+
+        subject.cloudwatch_client
+      end
+    end
+
+    describe 'when verbose is passed in' do
+      let(:opts) { { verbose: true } }
+      let(:logger) { double Logger }
+
+      before do
+        expect(Logger).to receive(:new).with(STDERR).and_return logger
+        default_args[:logger] = logger
+      end
+
+      it 'creates a client with the expected logger' do
+        expect(Reporting::CloudwatchClient).to receive(:new).with(default_args)
```

Comment:
My general preference is to check results (like check the `client.logger`) instead of stubbing `::new`

Reply:
for the purposes of this test i care that the values are being passed in as expected, so my preference is to check them as arguments here.